### ISSUE TITLE
Unable to Get Campaign Statistics.

### ISSUE THREAD

**Issue by kush-1277** [gophish/gophish#1108]
**Thanks for reaching out! We're happy to help resolve issues as quickly as possible.**

**Please use this template when creating a new issue. If you do not follow this template format, your issue may be closed without comment.**

What version of Gophish are you using?: Latest

**Brief description of the issue** I am unable to get campaigns statistics other than Email Sent.

**What are you expecting to see happen?** I wish to see all the stats.

**What are you seeing happen?**
I am not able to get the figures of how many people opened the mail or how many clicked the link in the email.

Also, when I hover on the link in email I see **http://192.168.1.1/?rid=oYJ4Xje** instead of **http://192.168.1.1/?track=oYJ4Xje**. 

I have tried my URL as 
1. my router's IP address.
2. A web-page that I need to show as landing page.  
3. My system's local IP address
4. 192.168.1.1

Also, I had changed the listen url in **config.json** to my router's ip address.Now it is back to localhost.

I have included Tracking Image in the email template and also I have followed the steps in the User Guide that is provided in the Application.

I have simulated GET API in POSTMAN and have observed that the **ip, longitude and latitude** fields remain empty. I understood that if we have ip address field data then we can get numbers in the GUI.

Please provide as many steps as you can to reproduce the problem:
I had just launched a campaign.

**Comment by kush-1277** [comment]
![12](https://user-images.githubusercontent.com/31572699/41221322-6df33446-6d81-11e8-925b-d9d84e6cc0d1.PNG)

This is the screenshot of the GoPhish terminal.

Thanks in Advance!

**Comment by jordan-wright** [comment]
So, there's a lot to unpack here.

It sounds like the core issue is that you aren't seeing events (clicks, opens, etc.) in Gophish.

> Also, when I hover on the link in email I see http://192.168.1.1/?rid=oYJ4Xje

So, a couple of things:

* Keep in mind that only people who can hit the router's internal IP address (192.168.1.1) will be able to hit this URL. If you want people outside your internal network to be able to reach your Gophish instance, you will need to use an external IP address or domain name pointing to your server.
* The `rid` format is expected and correct. There should be no `track` parameter.

So let's assume that you are on your internal network and you click that link. What do you see in your browser? Do you see a connection log show up in the Gophish terminal as soon as you click that link that contains `?rid=oYJ4Xje` in it?

**Comment by kush-1277** [comment]
No, I am not getting the connection log when I click on the link. I just get the refresh log of GET /api/... I can launch a campaign and share the entire log with you if it helps.

Also, while launching one of the campaigns I got errors like 

1. maillog.go:131: Erroring out result IOokoqf.
2. campaign.go:289: record not found.

These errors were for only one campaign and I guess they are irrelevant to the problem but I am not sure.

**Comment by jordan-wright** [comment]
What do you see in your browser when you click on the link?

You'll want to make sure that the IP address you're using the in the URL points to your Gophish server. If you're on a standard home network, `192.168.1.1` will probably be pointing to your router which isn't what you want. Instead, you'll need to your computer's IP address (on Windows you can do an `ipconfig` to find that out). However, that will be the internal address, so if you want people to hit your Gophish server from outside your network, you need to use an external IP address that points to your Gophish server (after you set up the correct networking to forward connections to your Gophish server).

**Comment by kush-1277** [comment]
Sorry, for the late reply.
I see the following screen when I click on the link.
This is my public ip address.
I have opened the link from Internal as well as external networks.

![issue1](https://user-images.githubusercontent.com/31572699/41700835-a9789768-7547-11e8-9315-f434ec5b4041.PNG)

**Comment by jordan-wright** [comment]
Hi @kush-1277,

Sorry for the delay in getting back to you. You're correct that this error indicates there is some network issue between your browser and the server running Gophish. I'm afraid that there isn't any help we can provide towards tracking that down, since it's highly dependent on your network layout.

You'll want to make sure you have all the proper routing and port forwarding set up to access Gophish externally. I'd recommend starting by making sure you can reach it internally, then working your way out.

Hope this helps!

**Comment by Sandeeppp1986** [comment]
https://github.com/jordan-wright, I am new to IT, i have configured Gophish to send email inside organisation. I am able to send email, But after opening Email or clicking on link, in campaign it is not showing up. I have attached screen shots, Please let me know if i am doing any mistakes. 
![image](https://user-images.githubusercontent.com/76405787/102721711-91dc1200-4322-11eb-81ec-44bfee4c3516.png)
![image](https://user-images.githubusercontent.com/76405787/102721736-ba640c00-4322-11eb-9574-e7f19bd3d6b6.png)
![image](https://user-images.githubusercontent.com/76405787/102721760-dbc4f800-4322-11eb-8f4f-1f32431bcf4d.png)
![image](https://user-images.githubusercontent.com/76405787/102721775-f5663f80-4322-11eb-8468-0abc770050c7.png)
![image](https://user-images.githubusercontent.com/76405787/102721806-1464d180-4323-11eb-83e9-32edaaa68efc.png)

**Comment by Sandeeppp1986** [comment]
Hi @jordan-wright  It will be great help for me, if you get some 5 min time from your busy schedules. Thanks on advance. :)
